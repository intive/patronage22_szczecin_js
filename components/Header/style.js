import styled from 'styled-components'

export const HeaderContainer = styled.header`
  position: sticky;
  left: 0;
  top: 0;
  width: 100%;
  height: 4.5rem;
  background: #FFFFFF;
  `

export const HeaderWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 100%;
  padding: 0 5%;

  @media (min-width: 1440px) {
    padding: 0 2.5rem;
  }
`

export const HeaderTitle = styled.a`
  display: flex;
  font-weight: 700;
  font-size: 1.25rem;
  line-height: 1.5rem;
  text-decoration: none;
  color: inherit;
  `

export const HeaderTitleText = styled.p`
  margin-left: 0.5rem;
  color: #222222;
`
