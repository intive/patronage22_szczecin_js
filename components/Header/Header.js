import { HeaderContainer, HeaderWrapper, HeaderTitle, HeaderTitleText } from './style'
import Icon from '../Icon/Icon'
import Link from 'next/link'

export default function Header (props) {
  return (
    <HeaderContainer>
      <HeaderWrapper>
        <Link href='/'>
          <HeaderTitle>
            <Icon name='crop_original' />
            <HeaderTitleText>{props.title}</HeaderTitleText>
          </HeaderTitle>
        </Link>
      </HeaderWrapper>
    </HeaderContainer>
  )
}
