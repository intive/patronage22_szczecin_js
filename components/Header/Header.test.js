/* globals describe, it, expect  */

import { render } from '@testing-library/react'
import Header from './Header'

describe('Header', () => {
  it('should render Header', () => {
    const { container } = render(<Header />)

    expect(container).toMatchSnapshot()
  })
})
