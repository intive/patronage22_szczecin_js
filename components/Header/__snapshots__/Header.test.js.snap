// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header 1`] = `
<div>
  <header
    class="sc-bdvvtL bdAKOj"
  >
    <div
      class="sc-gsDKAQ fjfUlk"
    >
      <a
        class="sc-dkPtRN ijCfmr"
        href="/"
      >
        <span
          class="material-icons"
        >
          crop_original
        </span>
        <p
          class="sc-hKwDye bcymKS"
        />
      </a>
    </div>
  </header>
</div>
`;
